src/extractor/guidance/turn_handler.cpp

```diff
@@ -476,108 +587,153 @@ std::pair<std::size_t, std::size_t> TurnHandler::findFork(const EdgeID via_edge,
+        {
+            if (base_index == compare_index)
             {
-                return isObviousOfTwo(via_edge, intersection[right + 1], intersection[right]) ||
-                       isObviousOfTwo(via_edge, intersection[right], intersection[right + 1]) ||
-                       isObviousOfTwo(via_edge, intersection[left], intersection[right + 1]) ||
-                       isObviousOfTwo(via_edge, intersection[right + 1], intersection[left]);
+                continue;
             }
-            return false;
-        }();
-
-        // A fork can only happen between edges of similar types where none of the ones is obvious
-        const bool has_compatible_classes = [&]() {
-            const bool ramp_class = node_based_graph.GetEdgeData(intersection[right].eid)
-                                        .road_classification.IsLinkClass();
-            for (std::size_t index = right + 1; index <= left; ++index)
-                if (ramp_class !=
-                    node_based_graph.GetEdgeData(intersection[index].eid)
-                        .road_classification.IsLinkClass())
-                    return false;
-
-            const auto in_classification =
-                node_based_graph.GetEdgeData(intersection[0].eid).road_classification;
-            for (std::size_t base_index = right; base_index <= left; ++base_index)
+            const auto compare_class =
+                node_based_graph.GetEdgeData(intersection[compare_index].eid).road_classification;
+            if (obviousByRoadClass(via_class, base_class, compare_class))
             {
-                const auto base_classification =
-                    node_based_graph.GetEdgeData(intersection[base_index].eid).road_classification;
-                for (std::size_t compare_index = right; compare_index <= left; ++compare_index)
-                {
-                    if (base_index == compare_index)
-                        continue;
-
-                    const auto compare_classification =
-                        node_based_graph.GetEdgeData(intersection[compare_index].eid)
-                            .road_classification;
-                    if (obviousByRoadClass(
-                            in_classification, base_classification, compare_classification))
-                        return false;
-                }
+                return false;
             }
-            return true;
-        }();
+        }
+    }
+    return true;
+}
 
-        // check if all entries in the fork range allow entry
-        const bool only_valid_entries = [&]() {
-            BOOST_ASSERT(right <= left && left < intersection.size());
+// Checks whether a three-way-intersection coming from `via_edge` is a fork
+// with `intersection` as described as in #IntersectionExplanation@intersection_handler.hpp
+TurnHandler::Fork TurnHandler::findFork(const EdgeID via_edge,
+                                        const Intersection &intersection) const
```

Comment:
What happens if `intersection` is empty (the UTurn road only)? Is size = 3 a pre-condition?